h2o-automl/src/main/java/ai/h2o/automl/AutoML.java

```diff
@@ -529,7 +529,7 @@ public Job trainModel(Key<Model> key, String algoURLName, Model.Parameters parms
     Model.Parameters defaults = builder._parms;
     builder._parms = parms;
 
-    setCommonModelBuilderParams(builder._parms);
+    setCommonModelBuilderParams(builder._parms); //TODO why we set it again after setting it in AutoMl.defaultRandomForest() method?
```

Comment:
This sets common build parameters for all automl models.

Reply:
Maybe I'm missing something.... I will try to explain my reasoning on AutoML.defaultRandomForest() example. I just noticed that same method is being called twice.

instead of doing this
````
    setCommonModelBuilderParams(drfParameters);

    drfParameters._stopping_tolerance = this.buildSpec.build_control.stopping_criteria.stopping_tolerance();

    Job randomForestJob = trainModel(null, "drf", drfParameters);

where inside trainModel we are doing the following:

   builder._parms = parms;
    setCommonModelBuilderParams(builder._parms);
````
So we are setting all the parameters twice..... although only one parameter `_stopping_tolerance`  is beeing changed.

Maybe we can do  it this way.... first change `_stopping_tolerance` and then call `setCommonModelBuilderParams` only once?
````
drfParameters._stopping_tolerance = this.buildSpec.build_control.stopping_criteria.stopping_tolerance();

 setCommonModelBuilderParams(drfParameters);
Job randomForestJob = trainModel(null, "drf", drfParameters); // remove setCommonModelBuilderParams call from trainModel
````